### PR TITLE
Update memcached client for better concurrency in metrics. 

### PR DESCRIPTION
**Pull request by drcrallen**
Also fixes another injection problem

Fixes https://github.com/druid-io/druid/issues/1574

### CODE REVIEW

**Comment by drcrallen** [comment]
Encountered NPE closing to investigate:

```
java.lang.NullPointerException
        at com.metamx.metrics.SysMonitor$TcpStats.emit(SysMonitor.java:553) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.metamx.metrics.SysMonitor.doMonitor(SysMonitor.java:124) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.metamx.metrics.AbstractMonitor.monitor(AbstractMonitor.java:41) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.metamx.metrics.MonitorScheduler$1.call(MonitorScheduler.java:121) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.metamx.metrics.MonitorScheduler$1.call(MonitorScheduler.java:116) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.metamx.common.concurrent.ScheduledExecutors$4.run(ScheduledExecutors.java:173) [druid-selfcontained-ee03e22.jar:ee03e22]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_45]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_45]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_45]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_45]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_45]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_45]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_45]
```

**Comment by drcrallen** [comment]
```
java.lang.NullPointerException
        at com.metamx.metrics.SysMonitor$TcpStats.emit(SysMonitor.java:553) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.metamx.metrics.SysMonitor.doMonitor(SysMonitor.java:124) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.metamx.metrics.AbstractMonitor.monitor(AbstractMonitor.java:41) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.metamx.metrics.MonitorScheduler$1.call(MonitorScheduler.java:121) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.metamx.metrics.MonitorScheduler$1.call(MonitorScheduler.java:116) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.metamx.common.concurrent.ScheduledExecutors$4.run(ScheduledExecutors.java:173) [druid-selfcontained-ee03e22.jar:ee03e22]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_45]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_45]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_45]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_45]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_45]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_45]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_45]


2015-09-22T02:08:21,431 ERROR [MonitorScheduler-0] com.metamx.common.concurrent.ScheduledExecutors - Uncaught exception.
java.lang.IllegalArgumentException: Multiple entries with same key: sys/tcp/state/established=277 and sys/tcp/state/established=277
        at com.google.common.collect.ImmutableMap.checkNoConflict(ImmutableMap.java:150) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.google.common.collect.RegularImmutableMap.checkNoConflictInBucket(RegularImmutableMap.java:104) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.google.common.collect.RegularImmutableMap.<init>(RegularImmutableMap.java:70) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.google.common.collect.ImmutableMap$Builder.build(ImmutableMap.java:254) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.metamx.metrics.SysMonitor$TcpStats.emit(SysMonitor.java:566) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.metamx.metrics.SysMonitor.doMonitor(SysMonitor.java:124) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.metamx.metrics.AbstractMonitor.monitor(AbstractMonitor.java:41) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.metamx.metrics.MonitorScheduler$1.call(MonitorScheduler.java:121) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.metamx.metrics.MonitorScheduler$1.call(MonitorScheduler.java:116) ~[druid-selfcontained-ee03e22.jar:ee03e22]
        at com.metamx.common.concurrent.ScheduledExecutors$4.run(ScheduledExecutors.java:173) [druid-selfcontained-ee03e22.jar:ee03e22]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_45]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_45]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_45]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_45]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_45]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_45]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_45]
```

I seem to be getting the error multiple times in TcpStats, but I'm not sure if they are related to this patch, so I'm reopening.

File: server/src/main/java/io/druid/client/cache/MemcachedCache.java [hunk 1]
```diff
@@ -273,15 +283,20 @@ public void addMeter(String name)
         @Override
         public void removeMeter(String name)
         {
```
Comment (nishantmonu51) [inline server/src/main/java/io/druid/client/cache/MemcachedCache.java]: seems like it will never remove the meter ?

Comment (drcrallen) [inline server/src/main/java/io/druid/client/cache/MemcachedCache.java]: Correct. Removes are ignored. I prevent memory leaks by limiting the set of meters allowed. to isInteresting

**Comment by drcrallen** [comment]
`java.lang.AssertionError: Wait for /test1 to be created`

**Comment by drcrallen** [comment]
```org.apache.zookeeper.KeeperException$NoNodeException: KeeperErrorCode = NoNode for /druid/loadqueue/localhost:1234/test_load_queue_peon_2014-10-31T00:00:00.000Z_2014-11-01T00:00:00.000Z_2015-05-27T03:38:35.683Z
	at org.apache.zookeeper.KeeperException.create(KeeperException.java:111)
	at org.apache.zookeeper.KeeperException.create(KeeperException.java:51)
	at org.apache.zookeeper.ZooKeeper.delete(ZooKeeper.java:873)
	at org.apache.curator.framework.imps.DeleteBuilderImpl$5.call(DeleteBuilderImpl.java:242)
	at org.apache.curator.framework.imps.DeleteBuilderImpl$5.call(DeleteBuilderImpl.java:236)
	at org.apache.curator.RetryLoop.callWithRetry(RetryLoop.java:107)
	at org.apache.curator.framework.imps.DeleteBuilderImpl.pathInForeground(DeleteBuilderImpl.java:233)
	at org.apache.curator.framework.imps.DeleteBuilderImpl.forPath(DeleteBuilderImpl.java:217)
	at org.apache.curator.framework.imps.DeleteBuilderImpl.forPath(DeleteBuilderImpl.java:41)
	at io.druid.server.coordinator.LoadQueuePeonTest.testMultipleLoadDropSegments(LoadQueuePeonTest.java:253)
```

File: server/src/main/java/io/druid/client/cache/MemcachedCache.java (hunk 1 above)
Comment (nishantmonu51) [inline server/src/main/java/io/druid/client/cache/MemcachedCache.java]: ok just wanted to make sure its intentional

**Comment by nishantmonu51** [comment]
+1

**Comment by xvrl** [comment]
:+1: btw :)

**Pull request merged**